lib/perl/Genome/InstrumentData/Command/Import/Basic.pm

```diff
@@ -112,72 +107,32 @@ sub execute {
 
-sub _resolve_original_format {
-    my $self = shift;
-    $self->status_message('Resolve original format...');
-
-    if ( $self->original_format ) {
-        $self->status_message('Original format: '.$self->original_format);
-        return $self->original_format;
-    }
-
-    my @source_files = $self->source_files;
-    my $helpers = Genome::InstrumentData::Command::Import::WorkFlow::Helpers->get;
-    my %formats;
-    my $is_archived = 0;
-    for my $source_file ( @source_files ) {
-        my $format = $helpers->source_file_format($source_file);
-        return if not $format;
-        $formats{$format}++;
-        $is_archived++ if $helpers->is_source_file_archived($source_file);
-    }
-
-    my @formats = keys %formats;
-    if ( @formats > 1 ) {
-        $self->error_message('Got more than one format when trying to determine format!');
-        return;
-    }
-
-    my $format = $formats[0];
-    if ( $is_archived > 1 ) {
-        $self->error_message('More than one file is archived. Please import them separately.');
-        return;
-    }
-    elsif ( $is_archived ) {
-        $format .= '_archive';
-    }
-    $self->status_message('Original format: '.$format);
-
-    my $max_source_files = ( $format =~ /^fast[aq]$/ ? 2 : 1 );
-    if ( @source_files > $max_source_files ) {
-        $self->error_message("Cannot handle more than $max_source_files source files!");
-        return;
-    }
-
-    $self->status_message('Resolve original format...done');
-    return $self->original_format($format);
-}
-
 sub _resolve_instrument_data_properties {
     my $self = shift;
 
     my @instrument_data_properties = $self->instrument_data_properties;
-    push @instrument_data_properties, 'description='.$self->description if defined $self->description;
-    push @instrument_data_properties, 'downsample_ratio='.$self->downsample_ratio if defined $self->downsample_ratio;
-    my $instdata_props_processor = Genome::InstrumentData::Command::Import::WorkFlow::ResolveInstDataProperties->execute(
+    my $instdata_props_processor = Genome::InstrumentData::Command::Import::WorkFlow::Inputs->create(
+        source_files => [ $self->source_files ],
         instrument_data_properties => \@instrument_data_properties,
+        description => ( $self->description || undef ),
+        downsample_ratio => ( $self->downsample_ratio || undef ),
     );
-    if ( not $instdata_props_processor->result ) {
-        $self->error_message('Failed to process instrument data properties!');
-        return;
-    }
 
-    my $instrument_data_properties =  $instdata_props_processor->resolved_instrument_data_properties;
-    if ( not $instrument_data_properties->{original_data_path} ) {
-        $instrument_data_properties->{original_data_path} = join(',', $self->source_files);
-    }
+    return $self->inputs($instdata_props_processor);
+}
 
-    return $self->_instrument_data_properties($instrument_data_properties);
+sub _die_if_original_data_path_previously_imported {
```

Comment:
It is true that one could start while another is running, and 2 copies could ultimately exist. There are 2 md5 checks, one at the beginning and one again at the end when the instdata are created that prevent it happening most of the time. Locking can be added, but that would be another PR.